gateways/checkoutdotcom/request_builder.go

```diff
@@ -0,0 +1,59 @@
+package checkoutdotcom
+
+import (
+	"github.com/BoltApp/sleet"
+	"github.com/checkout/checkout-sdk-go/common"
+	"github.com/checkout/checkout-sdk-go/payments"
+)
+
+func buildChargeParams(authRequest *sleet.AuthorizationRequest) *payments.Request {
+	var source = payments.CardSource{
+		Type: "card",
+		Number: authRequest.CreditCard.Number,
+		ExpiryMonth: uint64(authRequest.CreditCard.ExpirationMonth),
+		ExpiryYear: uint64(authRequest.CreditCard.ExpirationYear),
+		Name: authRequest.CreditCard.FirstName + " " + authRequest.CreditCard.LastName,
+		CVV: authRequest.CreditCard.CVV,
+		BillingAddress: &common.Address{
+			AddressLine1: *authRequest.BillingAddress.StreetAddress1,
```

Comment:
we are already passing in a pointer to authRequest. This would be a pointer to a pointer.

Reply:
Sorry, still a bit new to golang here. What are you asking me to change?

It seems that the StreetAddress1 is a string pointer and the AddressLine1 is looking for a string (defined by the checkout.com SDK). So I added the * to dereference the string pointer to be a string. When I try removing the *, I get

`Cannot use 'authRequest.BillingAddress.StreetAddress1' (type *string) as the type string`